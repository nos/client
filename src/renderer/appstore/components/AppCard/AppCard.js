import React from 'react';
import { func, bool } from 'prop-types';
import { noop } from 'lodash';

import styles from './AppCard.scss';
import appShape from '../../shape/appShape';

const AppCard = ({ app, openApp, internal }) => {
  const { title, blurb, url, img_url: imgUrl } = app;
  const clickHandler = internal ? noop : openApp(url);

  return (
    <div className={styles.appcard} role="button" tabIndex={0} onClick={clickHandler}>
      <div className={styles.imgContainer}>
        <img alt={`${title} logo`} src={imgUrl} />
      </div>
      <div className={styles.appInfo}>
        <h2>{title}</h2>
        <p>{blurb}</p>
      </div>
    </div>
  );
};

AppCard.propTypes = {
  app: appShape.isRequired,
  openApp: func.isRequired,
  internal: bool
};

AppCard.defaultProps = {
  internal: false
};

export default AppCard;
