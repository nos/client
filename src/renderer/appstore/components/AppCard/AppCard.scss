.appcard {
  background-color: transparent;
  box-shadow: 0px 2px 6px rgba(30, 27, 82, 0.12);
  border-radius: 4px;
  width: 380px;
  cursor: pointer;
  z-index: 0;

  &:hover {
    box-shadow: 2px 4px 6px rgba(30, 27, 82, 0.12);
  }

  .imgContainer {
    width: 380px;
    height: 266px;

    img {
      max-width: 380px;
      max-height: 266px;
      display: block;
      margin: auto;
      border-top-left-radius: 4px;
      border-top-right-radius: 4px;
    }
  }

  .appInfo {
    background-color: white;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    padding: 16px;

    @include theme(DARK) {
      @extend %panelColor;
    }

    h2 {
      font-family: $primary-font-stack;
      color: $brand;
      font-size: 16px;
      margin: 0 0 4px;
      font-weight: 500;
    }

    p {
      color: $gray-4;
      font-size: 12px;
      height: 32px;
      text-overflow: ellipsis;
      overflow: hidden;
      margin: 0;
      font-family: $primary-font-stack;

      @include theme(DARK) {
        color: $gray-2;
      }
    }
  }
}

@media (max-width: 900px) {
  .appbutton {
    width: 190px;

    .imgContainer {
      width: 190px;
      height: 133px;

      img {
        max-width: 190px;
        max-height: 133px;
      }
    }
  }
}
