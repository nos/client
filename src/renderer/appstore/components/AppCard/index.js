import { connect } from 'react-redux';

import { openTab } from 'browser/actions/browserActions';

import AppCard from './AppCard';

const mapDispatchToProps = (dispatch) => ({
  openApp: (target) => () =>
    dispatch(
      openTab({
        target
      })
    )
});

export default connect(null, mapDispatchToProps)(AppCard);
