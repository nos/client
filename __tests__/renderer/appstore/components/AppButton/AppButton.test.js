import React from 'react';
import { mount } from 'enzyme';

import AppCard from 'appstore/components/AppCard/AppCard';

describe('<AppCard />', () => {
  const defaultProps = {
    app: {
      title: 'mockTitle',
      blurb: 'mockBlurb',
      url: 'mockUrl',
      img_url: 'mockImgUrl'
    },
    openApp: jest.fn()
  };

  it('renders correctly', () => {
    const wrapper = mount(<AppCard {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
