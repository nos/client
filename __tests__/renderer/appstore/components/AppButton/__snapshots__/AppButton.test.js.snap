// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppButton /> renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <AppButton
    app={
      Object {
        "blurb": "mockBlurb",
        "img_url": "mockImgUrl",
        "title": "mockTitle",
        "url": "mockUrl",
      }
    }
    openApp={
      [MockFunction] {
        "calls": Array [
          Array [
            "mockUrl",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "app": Object {
        "blurb": "mockBlurb",
        "img_url": "mockImgUrl",
        "title": "mockTitle",
        "url": "mockUrl",
      },
      "openApp": [MockFunction] {
        "calls": Array [
          Array [
            "mockUrl",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="appbutton"
        role="button"
        tabindex="0"
      >
        <div
          class="imgContainer"
        >
          <img
            alt="mockTitle logo"
            src="mockImgUrl"
          />
        </div>
        <div
          class="appInfo"
        >
          <h2>
            mockTitle
          </h2>
          <p>
            mockBlurb
          </p>
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="imgContainer"
          >
            <img
              alt="mockTitle logo"
              src="mockImgUrl"
            />
          </div>,
          <div
            className="appInfo"
          >
            <h2>
              mockTitle
            </h2>
            <p>
              mockBlurb
            </p>
          </div>,
        ],
        "className": "appbutton",
        "onClick": undefined,
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="imgContainer"
          >
            <img
              alt="mockTitle logo"
              src="mockImgUrl"
            />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="mockTitle logo"
              src="mockImgUrl"
            />,
            "className": "imgContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <img
                alt="mockTitle logo"
                src="mockImgUrl"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "mockTitle logo",
                "src": "mockImgUrl",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="appInfo"
          >
            <h2>
              mockTitle
            </h2>
            <p>
              mockBlurb
            </p>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                mockTitle
              </h2>,
              <p>
                mockBlurb
              </p>,
            ],
            "className": "appInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h2>
                mockTitle
              </h2>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mockTitle",
              },
              "ref": null,
              "rendered": Array [
                "mockTitle",
              ],
              "type": "h2",
            },
            Object {
              "instance": <p>
                mockBlurb
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mockBlurb",
              },
              "ref": null,
              "rendered": Array [
                "mockBlurb",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "app": Object {
          "blurb": "mockBlurb",
          "img_url": "mockImgUrl",
          "title": "mockTitle",
          "url": "mockUrl",
        },
        "openApp": [MockFunction] {
          "calls": Array [
            Array [
              "mockUrl",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="appbutton"
          role="button"
          tabindex="0"
        >
          <div
            class="imgContainer"
          >
            <img
              alt="mockTitle logo"
              src="mockImgUrl"
            />
          </div>
          <div
            class="appInfo"
          >
            <h2>
              mockTitle
            </h2>
            <p>
              mockBlurb
            </p>
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="imgContainer"
            >
              <img
                alt="mockTitle logo"
                src="mockImgUrl"
              />
            </div>,
            <div
              className="appInfo"
            >
              <h2>
                mockTitle
              </h2>
              <p>
                mockBlurb
              </p>
            </div>,
          ],
          "className": "appbutton",
          "onClick": undefined,
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="imgContainer"
            >
              <img
                alt="mockTitle logo"
                src="mockImgUrl"
              />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="mockTitle logo"
                src="mockImgUrl"
              />,
              "className": "imgContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <img
                  alt="mockTitle logo"
                  src="mockImgUrl"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "mockTitle logo",
                  "src": "mockImgUrl",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="appInfo"
            >
              <h2>
                mockTitle
              </h2>
              <p>
                mockBlurb
              </p>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  mockTitle
                </h2>,
                <p>
                  mockBlurb
                </p>,
              ],
              "className": "appInfo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h2>
                  mockTitle
                </h2>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "mockTitle",
                },
                "ref": null,
                "rendered": Array [
                  "mockTitle",
                ],
                "type": "h2",
              },
              Object {
                "instance": <p>
                  mockBlurb
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "mockBlurb",
                },
                "ref": null,
                "rendered": Array [
                  "mockBlurb",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
